REFRAME.md

```diff
@@ -280,6 +282,45 @@ Response:
 {"PutIPNSResponse : {}"}
 ```
 
+#### GetPeerAddresses
+
+A message for getting the addresses of a libp2p peer
+
+```ipldsch
+    type GetPeerAddressesRequest struct {
+        ID Bytes # libp2p PeerID
+        RecordTypes [String]
+    }
+
+    type Multiaddresses [Bytes] # Each element in the list is the binary representation of a complete multiaddr without a peerID suffix
+    type Libp2pSignedPeerRecord Bytes # https://github.com/libp2p/specs/blob/8c967a22bfbaff1ae41072b358fdba7c5883b6a4/RFC/0003-routing-records.md
+
+    type PeerRecordType union {
+     | Multiaddresses "multiaddrs"
+     | Libp2pSignedPeerRecord "769" // the libp2p signed peer record entry interpreted as decimal https://github.com/multiformats/multicodec/blob/f5dd49f35b26b447aa380e9a491f195fd49d912c/table.csv#L129
+     | Any default
+    } representation keyed
+    
+    type GetPeerAddressesResponse struct {
+        Records [PeerRecordType]

```

Comment:
Does it make sense to return `[PeerRecordType]` here rather than just a single `PeerRecordType` since we're allowed to send back a stream of `GetPeerAddressesResponse`? Probably doesn't hurt, but we might want to build some conventions here.

Reply:
I'd keep the array. A receiver might want to forward/use results as soon as the first one arrives. They have no good way of knowing that more are coming back to back.